### PR TITLE
[WIP] Bump SGLang to 0.5.12 and fix smoke regressions

### LINKED ISSUE #658: Upgrade the pinned SGLang backend (0.5.8 → latest)
## Motivation

https://github.com/sgl-project/sglang-omni/issues/441

We pin `sglang==0.5.8` (Jan 2026); upstream is now 0.5.12. Two reasons to upgrade.

 Omni currently runs only on Hopper and Ampere. B-cards are becoming the common deployment target in the US and we don't support them yet. Newer SGLang ships the SM100 path we'd need: FA4 with Blackwell support, multimodal attention defaulting to FA4 on SM100, and TRT-LLM DSA kernels default on SM100.

Upstream has reworked the multimodal/encoder path (stronger transformers/VLM modeling backend). Much of our encoder-side glue is hand-rolled against 0.5.8 internals; the newer APIs should let us delete some of it and make future encoder work easier.

## Pinned / gated to 0.5.8 today (must move together)

- `pyproject.toml` L29 — `sglang==0.5.8`
- `model_runner/_sglang_qwen3_vl_patches.py` L29 — a hard allowlist `{"0.5.8","0.5.8.post1"}` that rejects any other version at runtime; it guards Qwen3-VL vision-layout patches that must be re-validated against the new layout.

## Version-tied debt to clear during the upgrade

- `models/qwen3_omni/components/talker.py` L59 — FIXME: dense code-predictor MLP quantization changed in newer SGLang.
- `models/qwen3_omni/config.py` L19 — FIXME: DeepGEMM compile workaround.
- `model_runner/model_worker.py` L378 — mirrors 0.5.8 FP8 CUTLASS MoE assertions; must be re-synced against upstream.
- `models/qwen3_asr/request_builders.py` L190/L210 — stashes fields in `MultimodalDataItem` assuming 0.5.8 has no `model_specific_data`; check whether newer SGLang formalizes it.
- `models/qwen3_omni/hf_config.py` L23 — relies on 0.5.8 MRotaryEmbedding selection.

## Coupled constraint

SGLang ≥0.5.10 moves to transformers 5.x, but we pin `transformers<5.0` (pyproject). The upgrade forces a transformers bump too; models need re-validation against it.

## Scope

- [ ] Bump the pin + allowlist; existing CI green on Hopper/Ampere (no regression).
- [ ] Bring up and add CI coverage on a Blackwell Machine (really?).
- [ ] Replace hand-rolled encoder glue with the new upstream encoder APIs where available.
- [ ] Resolve the FIXMEs above; drop workarounds the upgrade makes unnecessary.

### PR DESCRIPTION
**Pull request by albaNnaksqr**
## Motivation

Part of #658.

This PR moves the local SGLang Omni stack toward `sglang==0.5.12` and fixes the first set of compatibility regressions found during installation, unit testing, and local model smoke tests.

I noticed #677 and #679 are also working on the dependency bump, so this is intentionally positioned as supplemental runtime validation/fix work rather than a complete resolution of #658.

## Modifications

- Bump the pinned runtime stack:
  - `sglang==0.5.12`
  - `transformers==5.6.0`
  - `torch==2.11.0`
  - matching `torchvision`, `torchaudio`, and `torchcodec` versions.
- Restrict the Qwen3-VL patch allowlist to the revalidated SGLang `0.5.12` baseline.
- Update Qwen3-ASR request building for SGLang 0.5.12 multimodal behavior:
  - use inclusive multimodal offsets;
  - pass `feature_attention_mask` through `model_specific_data`.
- Remove the old Qwen3-Omni talker dense-MLP quantization workaround tied to SGLang 0.5.8.
- Fix unit-level compatibility regressions in:
  - Scheduler request-limit fixtures;
  - Higgs TTS request-builder timing test;
  - MOSS TTS CPU/unit-test sampling paths;
  - Ming legacy flash-attn symbol detection;
  - Qwen3 TTS Transformers 5 compatibility.
- Add Qwen3 TTS compatibility handling for Transformers 5 changes around:
  - `check_model_inputs`;
  - default RoPE initialization;
  - eager decode fallback when torch compile is disabled.
- Allow Qwen3 TTS engine startup to pass SGLang server-args overrides, which was needed for constrained-memory local smoke testing.

## Validation

Unit tests:

```bash
.venv-sglang-upgrade/bin/python -m pytest tests/unit_test -q
```

Result:

```text
1014 passed, 24 warnings
```

Local model smoke tests:

- Qwen3-ASR:
  - model: `Qwen3-ASR-1.7B`
  - result: request completed successfully and returned expected transcription-style text.
- Qwen3-TTS:
  - model: `Qwen3-TTS-12Hz-1.7B-Base`
  - endpoint: `/v1/audio/speech`
  - result: request completed successfully and produced a valid WAV response.
- Qwen3-Omni:
  - model: `Qwen3-Omni-30B-A3B-Instruct`
  - endpoint: `/v1/chat/completions`
  - result: text-only chat request completed successfully.

## Remaining Work

This PR does not claim to fully close #658. Remaining validation/work includes:

- broader Qwen3-Omni text/audio/video e2e validation;
- talker audio generation quality / WER checks;
- benchmark-style checks such as MMMU, MMSU, Video-MME, and Video-AMME where applicable;
- FP8 runtime validation if support is expected to remain unchanged;
- Blackwell CI coverage, if available.

## Related Issues

Refs #658.

### CODE REVIEW

**Comment by Hayden727** [comment]
Thank you for your contribution. Your proposal is significantly more persuasive. We will conduct a full version bump in the same PR later. We will invite you to test your platform at that time, and we will also add you as a co-author.

Furthermore, this is an important but not urgent issue for this project; it's a long-term plan, so there's no rush to implement it. It's something that needs to be considered after the project has stabilized and a new version is released.

**Comment by albaNnaksqr** [comment]
> Thank you for your contribution. Your proposal is significantly more persuasive. We will conduct a full version bump in the same PR later. We will invite you to test your platform at that time, and we will also add you as a co-author.
> 
> Furthermore, this is an important but not urgent issue for this project; it's a long-term plan, so there's no rush to implement it. It's something that needs to be considered after the project has stabilized and a new version is released.

Sounds good. I will keep this PR as a draft reference for the runtime fixes and local smoke-test results. When the full version bump resumes, I’m happy to help test on my platform.

**Comment by JingwenGu0829** [comment]
Thanks for the pr! We just moved the work to https://github.com/sgl-project/sglang-omni/pull/698 for better collaboration on the main branch instead of fork. 

We added you as co-author for #698 thanks to this pr's contribution. We will run all the end to end pipeline tests and further investigations in #698. Thanks again for the contribution.
